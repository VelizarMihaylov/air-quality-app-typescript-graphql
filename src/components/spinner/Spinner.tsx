import React from 'react'

import styled from 'styled-components'

export type SpinnerProps = {
  className?: string
}
const SpinnerComponent = styled.div`
  border: 8px solid ${({ theme }) => theme.colours.white};
  border-top: 8px solid ${({ theme }) => theme.colours.purple};
  border-radius: 50%;
  width: ${({ theme }) => theme.spacing(1.5)};
  height: ${({ theme }) => theme.spacing(1.5)};
  animation: spin 2s linear infinite;

  @keyframes spin {
    0% {
      transform: rotate(0deg);
    }

    100% {
      transform: rotate(360deg);
    }
  }
`
export const Spinner: React.FC<SpinnerProps> = ({
  className
}): React.ReactElement => {
  return <SpinnerComponent className={className} />
}
