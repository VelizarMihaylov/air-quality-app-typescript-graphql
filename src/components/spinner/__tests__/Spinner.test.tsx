import React from 'react'
import { create } from 'react-test-renderer'

import { Spinner } from '../Spinner'

import { ThemeProvider } from 'styled-components'
import generateTheme from 'theme'

describe('Spinner', () => {
  it('should render without throwing an error', () => {
    const renderer = create(
      <ThemeProvider theme={generateTheme()}>
        <Spinner />
      </ThemeProvider>
    )

    expect(renderer.toJSON()).toMatchSnapshot()
  })
})
