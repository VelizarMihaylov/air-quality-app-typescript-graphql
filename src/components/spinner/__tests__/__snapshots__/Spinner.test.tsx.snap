// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner should render without throwing an error 1`] = `
<div
  className="sc-bdfBwQ hGa-dgG"
/>
`;
